src/main/java/org/gentoo/java/ebuilder/maven/MavenCache.java

```diff
@@ -43,13 +43,13 @@ public String getDependency(final String groupId, final String artifactId,
         final Map<String, List<CacheItem>> artifactIds = cache.get(groupId);
 
         if (artifactIds == null) {
-            return null;
+            return "!!!groupId-not-found!!!";
```

Comment:
both these changes affect the behavior of MavenEbuilder. when it receives null value, it skips the dependency in dependencies declaration and mentions that in related comments where are dependencies that were impossible to resolve are mentioned. when returning string instead of null, the returned string will be placed in the dependencies without mentioning this fact in the related comment which makes the result ambiguous. i suggest you modify MavenEbuilder instead to enhance the info reported in the comments with information for what reason the match failed (if that is ever needed). you can enhance MavenCache for that to be able to query the cache for specific groupId or groupId:artifactId coordinates, something like hasGroupId(groupId) and hasArtifactId(groupId, artifactId).